stups-spring-oauth2-server/src/main/java/org/zalando/stups/oauth2/spring/server/TokenInfoResponseErrorHandler.java

```diff
@@ -1,16 +1,14 @@
 package org.zalando.stups.oauth2.spring.server;
 
-import static org.springframework.http.HttpStatus.BAD_REQUEST;
-
-import java.io.IOException;
-
 import org.springframework.http.client.ClientHttpResponse;
 import org.springframework.web.client.DefaultResponseErrorHandler;
 
+import java.io.IOException;
+
 class TokenInfoResponseErrorHandler extends DefaultResponseErrorHandler {
     @Override
     public void handleError(ClientHttpResponse response) throws IOException {
-        if (response.getStatusCode() != BAD_REQUEST) {
+        if (!response.getStatusCode().is4xxClientError()) {
```

Comment:
I would not "whitelist" every client error, because this will hide bugs in our implementation (e.g. "405 Method not Allowed"). Instead I would prefer a configurable set of status codes that mark a token as invalid.